File: swift/allergies/Sources/Allergies/Allergies.swift
struct Allergies {
    
    let score: UInt
    
    init(_ score: UInt) {
        self.score = score
    }
    
    func hasAllergy(_ allergen: Allergen) -> Bool {
        return self.score & allergen.rawValue > 0
    }

    enum Allergen: UInt {
        case eggs = 1
        case peanuts = 2
        case shellfish = 4
        case strawberries = 8
        case tomatoes = 16
        case chocolate = 32
        case pollen = 64
        case cats = 128
    }
}

File: kotlin/allergies/src/main/kotlin/Allergies.kt
class Allergies {
    var score: Int

    constructor(score: Int) {
        this.score = score
    }

    fun getList(): List<Allergen> {
        return Allergen.values().filter { isAllergicTo(it) }
    }

    fun isAllergicTo(allergen: Allergen): Boolean {
        return (allergen.score.and(score)) != 0
    }
}
